src/outside/grammar_transformation.cc

```diff
@@ -0,0 +1,967 @@
+/* {{{
+
+    This file is part of gapc (GAPC - Grammars, Algebras, Products - Compiler;
+      a system to compile algebraic dynamic programming programs)
+
+    Copyright (C) 2011-2023  Stefan Janssen
+         email: stefan.m.janssen@gmail.com or stefan.janssen@computational.bio.uni-giessen.de
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+}}} */
+
+#include <utility>
+#include "grammar_transformation.hh"
+#include "../expr/new.hh"
+
+bool Grammar::check_outside_parse_empty_word() {
+  if (this->ast.outside_generation()) {
+    for (std::vector<Yield::Size>::const_iterator i =
+         this->axiom->multi_ys().begin();
+         i != this->axiom->multi_ys().end(); ++i) {
+      if ((*i).low() > 0) {
+        std::ostringstream msg;
+        msg << "The minimal yield size of your grammar '" << *this->name
+            << "' is ";
+        (*i).low().put(msg);
+        msg << ", i.e. it cannot parse the empty input string ''."
+            << " For outside grammar generation, this means you are lacking a"
+            << " recursion basis which will result in empty results for "
+            << "ALL outside candidates! Try adding an alternative like "
+            << "nil(EMPTY) to your axiom.";
+        Log::instance()->warning(this->location, msg.str());
+        return false;
+      }
+    }
+  }
+  return true;
+}
+
+void Grammar::check_outside_requested_nonexisting_nts() {
+  /* double check that all NTs do exist that the user requested to
+   * be reported in the outside grammar.
+   */
+
+  if (!this->ast.get_outside_nt_list()) {
+    // the user did not request any outside NT to be reported
+    return;
+  }
+  assert(this->ast.get_outside_nt_list());
+
+  // a list that collects the names of non existent NTs
+  std::list<std::string> *warn_missing_nts = new std::list<std::string>();
```

Comment:
The `warn_missing_nts` vector is not properly deleted at the end the function, creating a small memory leak. You could either just stack-allocate it (I don't see a reason why this needs to be heap-allocated with `new` here) or add `delete warn_missing_nts` at the end of the function to avoid this. 

Or we let the OS clean up the memory after GAP-C exits, which we rely on anyways in other areas of the compiler, but obviously isn't really the proper way to clean up memory.

Reply:
I'd prefer to delete at the end ... because debugging via Eclipse / gdb often stalls with stack-allocations (not sure why)

Reply:
That would be perfectly fine as well.